utils/user_provided_param.hh

```diff
@@ -0,0 +1,117 @@
+#pragma once
+
+/*
+ * Copyright (C) 2024-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include <fmt/format.h>
+
+#include <seastar/util/bool_class.hh>
+#include <seastar/core/sstring.hh>
+
+#include "enum_set.hh"
+
+using namespace seastar;
+
+namespace utils {
+
+enum class optional_param_flag {
+    user_provided,
+};
+
+using optional_param_flags_set = enum_set<super_enum<optional_param_flag,
+    optional_param_flag::user_provided
+>>;
+
+template <class T = sstring>
+class optional_param_base {
```

Comment:
Do we really need super_enum for flags here, why not just have `bool _user_provided`?

Reply:
We have more than one flag.
Why does it matter?

Reply:
I wonder if I missed something. It's not a big issue.

Reply:
It's easier for the implementation for multiple flags (especially if we add more flags in the future), for string formatting, for passing to the constructor, etc.

Reply:
I don't see how, since you don't operate on flags independently of optional_param. Formatting of flags is only needed when formatting optional_param, and flags are not passed around, only set with setters.